rest_framework_json_api/utils.py

```diff
@@ -76,7 +79,11 @@ def get_resource_name(context):
     except AttributeError:
         try:
             serializer = view.get_serializer_class()
-            return get_resource_type_from_serializer(serializer)
+            if issubclass(serializer, PolymorphicModelSerializer) and \
```

Comment:
This is minor, but could we switch the order of these checks? It would be slightly more performant to check the boolean `expand_polymorphic_types` first and only do the subclass check if necessary since `and` will short circuit if the boolean is `False`.